examples/azure/terraform-azure-vm-example/main.tf

```diff
@@ -0,0 +1,161 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY AN ADVANCED AZURE VIRTUAL MACHINE
+# This is an advanced example of how to deploy an Azure Virtual Machine in an availability set, managed disk 
+# and networking with a public IP.
+# ---------------------------------------------------------------------------------------------------------------------
+# See test/azure/terraform_azure_vm_example_test.go for how to write automated tests for this code.
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "azurerm" {
+  version = "~> 2.29"
+  features {}
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# PIN TERRAFORM VERSION TO >= 0.12
+# The examples have been upgraded to 0.12 syntax
```

Comment:
NIT: given that the comment block below already explains the TF version, fix this out-of-date one. 

```suggestion
# PIN TERRAFORM VERSION
```

Reply:
Updated

Reply:
Updated